Archiver/src/Archiver.cpp

```diff
@@ -0,0 +1,116 @@
+/*
+ *  Copyright 2019-2021 Diligent Graphics LLC
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence),
+ *  contract, or otherwise, unless required by applicable law (such as deliberate
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental,
+ *  or consequential damages of any character arising as a result of this License or
+ *  out of the use or inability to use the software (including but not limited to damages
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and
+ *  all other commercial damages or losses), even if such Contributor has been advised
+ *  of the possibility of such damages.
+ */
+
+#include "pch.h"
+#include <fstream>
+using namespace Diligent;
+
+int main(int argc, char* argv[])
+{
+    EngineEnvironment::Initialize(argc, argv);
+
+    auto pEnvironment     = EngineEnvironment::GetInstance();
+    auto pArchiveFactory  = pEnvironment->GetArchiveFactory();
+    auto pMemoryAllocator = pEnvironment->GetDeviceObjectReflection();
+
+    RefCntAutoPtr<IArchiver> pBuilder;
+    pArchiveFactory->CreateArchiver(pEnvironment->GetSerializationDevice(), &pBuilder);
+
+    auto const& OutputFilePath = pEnvironment->GetDesc().OuputFilePath;
+    auto const& InputFilePaths = pEnvironment->GetDesc().InputFilePaths;
+
+    for (auto const& Path : InputFilePaths)
+    {
+        std::ifstream  stream(Path);
+        nlohmann::json Json = nlohmann::json::parse(stream);
+
+        for (auto const& Signature : Json["ResourceSignatures"])
+        {
+            IPipelineResourceSignature* pSignature = nullptr;
+            Deserialize(Signature, pSignature, pMemoryAllocator);
```

Comment:
If these methods use environment, they must be its members, and not use the singleton.
Singletons must be avoided unless absolutely necessary. In this case there seems to be no reason to use singleton.